File: macOS SDK v4 Sample/AppConfig.m
//
//  AppConfig.m
//  macOS SDK v4 Sample
//
//  Created by Paddle on 19/12/2018.
//  Copyright © 2018 Paddle.com LTD. All rights reserved.
//

#import "AppConfig.h"

// ====================================================================
// PLEASE ENTER THESE VALUES BEFORE RUNNING THE APP.
//
NSString *const PAD_API_KEY = @"YOUR API KEY";
NSString *const PAD_VENDOR_ID = @"YOUR VENDOR ID";
NSString *const PAD_VENDOR_NAME = @"YOUR VENDOR NAME";
NSString *const PAD_SDK_PRODUCT_ID = @"YOUR MAIN PRODUCT ID";
NSString *const PAD_SDK_PRODUCT_NAME = @"YOUR MAIN PRODUCT NAME";
NSString *const PAD_SDK_SECOND_PRODUCT_ID = @"YOUR SECOND PRODUCT ID";
NSString *const PAD_SDK_SECOND_PRODUCT_NAME = @"YOUR SECOND PRODUCT NAME";
//
// ====================================================================

@implementation AppConfig

+ (PADProductConfiguration *)configurationForMainProduct
{
    PADProductConfiguration *config = [PADProductConfiguration configuration:PAD_SDK_PRODUCT_NAME
                                                                  vendorName:PAD_VENDOR_NAME];
    config.trialText = @"Thank you for trying out the Random Number Generator 500";
    config.localizedTrialText = NSLocalizedStringFromTableInBundle(@"Thank you for trying out the Random Number Generator 500", nil, [NSBundle mainBundle], nil);
    config.trialType = PADProductTrialTimeLimited;
    config.trialLength = @14;
    config.price = @10;
    config.currency = @"USD";
    config.imagePath = [NSBundle.mainBundle pathForResource:@"AppIcon" ofType:@"png"];

    return config;
}

+ (PADProductConfiguration *)configurationForSecondProduct
{
    PADProductConfiguration *config = [PADProductConfiguration configuration:PAD_SDK_SECOND_PRODUCT_NAME
                                                                  vendorName:PAD_VENDOR_NAME];
    config.trialText = @"Thank you for trying out the Second Product";
    config.localizedTrialText = NSLocalizedStringFromTableInBundle(@"Thank you for trying out the Second Product", nil, [NSBundle mainBundle], nil);
    config.trialType = PADProductTrialNone;
    config.price = @10;
    config.currency = @"USD";
    config.imagePath = [NSBundle.mainBundle pathForResource:@"AppIcon" ofType:@"png"];

    return config;
}

@end

File: Swift/macOS SDK v4 Swift Sample/AppConfig.swift
//
//  AppConfig.swift
//  macOS SDK v4 Swift Sample
//
//  Created by Louis Harwood on 26/05/2021.
//  Copyright © 2021 Paddle. All rights reserved.
//

import Foundation
import Paddle

class AppConfig: NSObject {
    static var PAD_API_KEY = ""
    static let PAD_VENDOR_ID = ""
    static let PAD_SDK_PRODUCT_ID = ""
    static let PAD_SDK_SECOND_PRODUCT_ID = ""

    static let PAD_SDK_PRODUCT_NAME = "YOUR MAIN PRODUCT NAME"
    static let PAD_SDK_SECOND_PRODUCT_NAME = "YOUR SECOND PRODUCT NAME"
    
    class func configurationForMainProduct() -> PADProductConfiguration {
        let mainProductConfig = PADProductConfiguration()
        mainProductConfig.trialText = "Thank you for trying out the Random Number Generator 500"
        mainProductConfig.localizedTrialText = NSLocalizedString("Thank you for trying out the Random Number Generator 500", tableName: nil, bundle: Bundle.main, value:"", comment: "")
        mainProductConfig.trialType = PADProductTrialType.timeLimited
        mainProductConfig.trialLength = 14;
        mainProductConfig.price = 10;
        mainProductConfig.currency = "USD";
        mainProductConfig.imagePath = Bundle.main.path(forResource: "AppIcon", ofType: "png")
        
        return mainProductConfig
    }

    class func configurationForSecondProduct() -> PADProductConfiguration {
        let secondProductConfig = PADProductConfiguration()
        secondProductConfig.trialText = "Thank you for trying out the Second Product"
        secondProductConfig.localizedTrialText = NSLocalizedString("Thank you for trying out the Second Product", tableName: nil, bundle: Bundle.main, value:"", comment: "")
        secondProductConfig.trialType = PADProductTrialType.none
        secondProductConfig.price = 10;
        secondProductConfig.currency = "USD";
        secondProductConfig.imagePath = Bundle.main.path(forResource: "AppIcon", ofType: "png")
        
        return secondProductConfig
    }
}
